### PR TITLE
fix: Fixes an Interface field argument regression 

### PR DESCRIPTION
**Pull request by kidunot89**
What does this implement/fix? Explain your changes.
---------------------------------------------------
This fixes a regression to #3102 / #3100 

Does this close any currently open issues?
------------------------------------------
closes: #3108 

Any other comments?
-------------------

Given the following code: 

```php
add_action( 'graphql_register_types', function() {

	register_graphql_interface_type( 'InterfaceWithArgs', [
		'fields' => [
			'fieldWithArgs' => [
				'type' => 'String',
				'args' => [
					'interfaceArg' => [
						'type' => 'String',
					],
				],
				'resolve' => function( $source, $args ) {
					return $args['arg'];
				}
			],
		]
	] );

	register_graphql_object_type( 'ObjectTypeImplementingInterfaceWithArgs', [
		'interfaces' => [ 'InterfaceWithArgs' ],
		'fields' => [
			'fieldWithArgs' => [
				'type' => 'String',
				'resolve' => function() {
					return 'object value';
				}
			],
		],
	] );

	register_graphql_field( 'RootQuery', 'interfaceArgsTest', [
		'type' => 'ObjectTypeImplementingInterfaceWithArgs',
		'resolve' => function() {
			return true;
		},
	]);

});
```

### Before

The schema would be invalid:

CleanShot 2024-04-26 at 11 33 56

And this query would be invalid:

CleanShot 2024-04-26 at 11 35 38

### After

The schema loads as valid and the test field can be queried

CleanShot 2024-04-26 at 11 35 05

### CODE REVIEW

**Comment by justlevine** [comment]
@kidunot89 is this addressing something different than #3109 or is it just a different approach?

**Comment by jasonbahl** [comment]
@kidunot89 this breaks most of the workflows: https://github.com/wp-graphql/wp-graphql/pull/3112/checks

**Comment by kidunot89** [comment]
@jasonbahl @justlevine The breaking workflows are because of WPGraphQL TestCase. It should be resolved in the 3.0 release which I'm working on right now. I'll get it up and update the PR.

**Comment by justlevine** [comment]
> @jasonbahl @justlevine The breaking workflows are because of WPGraphQL TestCase. It should be resolved in the 3.0 release which I'm working on right now. I'll get it up and update the PR.

Do we need update wp-graphql-test-case in a bugfix PR?

Also, I'm still unclear how this differs in approach (or purpose?) to #3109 .

I'm happy to give the same sort of code review (inline docs and reduced cyclomatic complexity), but it would be helpful to know what (if anything) I should be comparing.

**Comment by jasonbahl** [comment]
> Do we need update wp-graphql-test-case in a bugfix PR?

@kidunot89 ya, I think we should handle any updates needed for WPGraphQL Test Case separately. Let's focus on the specific issue at hand here, a regression to the merging of Interface Fields with Object Type fields. The other changes are creating side effects that maybe should ultimately be handled, but not within this PR.

**Comment by jasonbahl** [comment]
@kidunot89 I added [another test](https://github.com/wp-graphql/wp-graphql/pull/3112/files#diff-b9b0ab687b2170fedfb7568ae8b1ae59ca53beec92447cba4fff22a2dbd25c09R635) to cover complex arg types, such as: 

```php
	'args' => [
		'nonNullableArg' => [
			'type' => [ 'non_null' => 'String' ],
			'defaultValue' => 'nonNullableArg',
		],
		'listOfArg' => [
			'type' => [ 'list_of' => 'String' ],
			'defaultValue' => [ 'listOfArg', 'listOfArg 2' ],
		],
		'nonNullListOfString' => [
			'type' => [ 'non_null' => [ 'list_of' => 'String' ] ],
			'defaultValue' => [ 'nonNullListOfString', 'nonNullListOfString 2' ],
		],
		'listOfNonNullString' => [
			'type' => [ 'list_of' => [ 'non_null' => 'String' ] ],
			'defaultValue' => [ 'listOfNonNullString', 'listOfNonNullString 2' ],
		],
	],
```

**Review by jasonbahl — APPROVED** [review]

File: tests/wpunit/InterfaceTest.php [hunk 1]
```diff
@@ -93,30 +93,24 @@ public function testObjectTypeInheritsInterfaceFields() {
...
-				'query' => $query,
-			]
-		);
-
-		codecept_debug( $actual );
+				id
+				testInt
+				testString
+				interfaceOnlyField
+        	}
```
Comment (justlevine) [inline tests/wpunit/InterfaceTest.php:102]:
```suggestion
					}
```

**Comment by justlevine** [comment]
@jasonbahl I'm going to open up a follow up PR to add some code quality improvements and add the [failure cases](https://coveralls.io/jobs/141583227/source_files/25212474721)

**Pull request merged**